### PR TITLE
#133 added "page not found" response

### PR DESCRIPTION
**Pull request by pinaf**
For #133
- added class `RsNotFound`
- in case of `NotFoundException`, respond with `RsNotFound`
- added puzzle to make not found page have more information

### CODE REVIEW

**Comment by ghost** [comment]
@pinaf thanks.

**Comment by pinaf** [comment]
@k0r np

**Comment by karato** [comment]
I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by karato** [comment]
@darkled review it

**Comment by asaen** [comment]
@rultor merge

**Comment by yegor256** [comment]
@pinaf it's a bad design. upgrade takes to 0.13 and use new version of `TkFallback`: http://www.takes.org/apidocs-0.13/org/takes/facets/fallback/package-summary.html

**Comment by pinaf** [comment]
@yegor256 fixed. note that I need access to the exception in order to keep the "error" functionality to that one had to be a pure `Fallback`

**Comment by pinaf** [comment]
@yegor256 I cannot test the last fallback (exception that is not 404) and that caused cobertura to complain about overall project line coverage, so I added an exclude for a file that has no tests at all `Launch`.

File: src/main/java/com/nerodesk/takes/TkApp.java [hunk 1]
```diff
@@ -202,18 +207,29 @@ public void run() {
...
+                new FbStatus(404, new TkFixed(new RsNotFound())),
+                new Fallback() {
+                    @Override
+                    public Iterator<Response> route(final RqFallback req) {
+                        final String exc = ExceptionUtils.getStackTrace(
+                            req.throwable()
+                        );
+                        Logger.info(this, "Exception thrown\n%s", exc);
+                        return Collections.singleton(
+                            Response.class.cast(
```
Comment (yegor256) [inline src/main/java/com/nerodesk/takes/TkApp.java]: it's better to do `Collection.<Response>singleton(...`

File: src/main/java/com/nerodesk/takes/RsNotFound.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.takes.rs.RsWrap;
+
+/**
+ * Response for 404 / Page not found.
+ *
+ * @author Felipe Pina (felipe.pina@protonmail.com)
+ * @version $Id$
+ * @since 0.4
+ */
+final class RsNotFound extends RsWrap {
```
Comment (yegor256) [inline src/main/java/com/nerodesk/takes/RsNotFound.java:42]: @pinaf I would rename it to `FbNotFound` and make it `implements Fallback` or `extends FbWrap`. This way you will get access to the exception being thrown.

**Comment by yegor256** [comment]
@pinaf now it's way better, see my comments above

File: src/main/java/com/nerodesk/takes/TkApp.java (hunk 1 above)
Comment (pinaf) [inline src/main/java/com/nerodesk/takes/TkApp.java]: @yegor256 good catch! fixing.

**Comment by pinaf** [comment]
@yegor256 done. take not that the exception I wanted to use was not in the "not found" case but in the catch all falllback. So I added `FbError`.

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@darkled I just added **16 mins** to your account, many thanks for your contribution.. 55124381 spent here; you're getting extra minutes here (c=1); +16 added to your rating, current score is: [+2489](http://www.netbout.com/b/35091?open=rating)

**Comment by karato** [comment]
@rultor deploy

**Pull request merged**